frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java

```diff
@@ -837,6 +850,97 @@ public boolean isSnapshotDisabled() {
+            e.printStackTrace();
+        }
+        tokenAuthorizationEnabled = true;
+    }
+
+    public void dumpKeyLogs() {
+        String managementKey = "";
+        String inferenceKey = "";
+        String apiKey = "";
+        try {
+            Method method = tokenClass.getMethod("getManagementKey");
+            managementKey = (String) method.invoke(tokenObject);
+            method = tokenClass.getMethod("getInferenceKey");
+            inferenceKey = (String) method.invoke(tokenObject);
+            method = tokenClass.getMethod("getKey");
+            apiKey = (String) method.invoke(tokenObject);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            e.printStackTrace();
+        }
+
+        logger.info("KEY FILE PATH: " + System.getProperty("user.dir") + "/key_file.txt");
+        logger.info("MANAGEMENT KEY: " + managementKey);
+        logger.info("INFERNCE KEY: " + inferenceKey);
+        logger.info("API KEY: " + apiKey);
+        logger.info(
+                "MANAGEMENT API Example: curl http://localhost:8081/models/<model> -H \"Authorization: Bearer "
+                        + managementKey
+                        + "\"");
+        logger.info(
+                "INFERNCE API Example: curl http://127.0.0.1:8080/predictions/<model> -T <examples/image_classifier/kitten.jpg> -H \"Authorization: Bearer "
+                        + inferenceKey
+                        + "\"");
+        logger.info(
+                "API API Example: curl localhost:8081/token?type=management -H \"Authorization: Bearer "
+                        + apiKey
+                        + "\"");
+    }
+
+    public boolean isTokenEnabled() {
+        return tokenAuthorizationEnabled;
+    }
+
+    // Calls the checkTokenAuthorization function in the token plugin
+    // expects two inputs: the fullhttpRequest and an integer which is associated with the type
+    // 0: token api
+    // 1: management api
+    // 2: inference api
+    public void checkTokenAuthorization(FullHttpRequest req, Integer requestType)
+            throws ModelException {
+
+        if (tokenAuthorizationEnabled) {
+            try {
+                Method method =
+                        tokenClass.getMethod(
+                                "checkTokenAuthorization",
+                                io.netty.handler.codec.http.FullHttpRequest.class,
+                                Integer.class);
+                boolean result = (boolean) (method.invoke(tokenObject, req, requestType));
+                if (!result) {
+                    throw new InvalidKeyException(
+                            "Token Authenticaation failed. Token either incorrect, expired, or not provided correctly");
+                }
+                System.out.println("TOKEN AUTHORIZATION WORKED");
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                e.printStackTrace();
```

Comment:
If an error occurs the default should be access denied if authorization was enabled in the first place.